김현진_퍼블리싱 작업 완료

guswls1846 opened the pull request:
## Describe your PR 📝
- feat(readme): todo list 요구 명세 작성
  - 보이는 대로 작업하면 먼길을 돌거 같아 readme에 todo list 요구 명세서를 작성해봤습니다.
- 폴더 구조 생성
  - [FSD(Feature-Sliced Design)](https://feature-sliced.design/docs/get-started/overview)를 참고했습니다.
  - app: 애플리케이션 전체에 영향을 미치는 파일을 관리하는 곳입니다. 현재는 `global.css`, `reset.css`이 있습니다.
  - pages: 하위 레이어들을 조합하여 최종적인 뷰를 만들어내는 곳입니다. 현재는  todo-app 컴포넌트를 정의하고있습니다.
  - widgets: todo-app 내부에 있는 다양한 컴포넌트를 위젯 형태로 관리하는 곳입니다. `todo-header`, `todo-footer`, `todo-topbar`, `todo-bottombar`등이 있습니다.
- header, footer 퍼블 작업

## How to change 🛠
- client/public/readme 폴더에 요구 명세의 시각적 자료를 위한 스샷들을 관리합니다.

## Screenshot
<img width="869" alt="스크린샷 2024-06-07 오후 5 23 20" src="https://github.com/simtudy/vanilla-todo-app/assets/32049764/e494f350-e32f-4882-8e6a-db7fde35d6c5">

## Comments for future 🌱
- 빠진 요구 명세가 있다면 귀뜸 부탁드립니다 👂👂
- 생각보다 todo에 많은 기능들이 있어서 놀랬어요. 😱
- shadow dom 을 사용하니까 reset.css, global.css 같이 전역적으로 적용되어야할 스타일들이 공통적용이 안되서 각 컴포넌트에 같은 파일을 중복으로 넣어주고 있습니다 🥲 이게 맞는 방법인지 모르겠네요 ..

Commit:
feat(readme): todo list 요구 명세 작성

Commit:
modify: fsd 폴더 구조 생성

Commit:
feat: header 퍼블

Commit:
feat: footer 퍼블

Commit:
modify(readme): header, footer 체크

Commit:
modify: 폴더 이름 수정

Commit:
modify(readme): header, footer 체크

jjanmo commented:
> 생각보다 todo에 많은 기능들이 있어서 놀랬어요. 😱

`제대로된`이라는 표현이 맞을지 모르겠지만,, 
제대로된 TodoApp을 만든다고 가정하면 들어 있는, 혹은 들어갈 수 있는 기능들이 많더라구요 ㅎㅎ

jjanmo commented on hyunjin/client/README.md:
```diff
@@ -0,0 +1,66 @@
+
+#### [예시 사이트](https://todomvc.com/examples/javascript-es6/dist/#/)
+
+- [x] 헤더 퍼블
```
(궁금) 적어주신 `퍼블` 이 무슨 의미인가요??

guswls1846 replied:
> (궁금) 적어주신 `퍼블` 이 무슨 의미인가요??

보통 기능 없이 디자인적인 요소만 따로 때놓고 작업하고 싶을때가 있는데 그런의미로 퍼블리싱에서 앞 두글잘만 자른겁니다 🤣

jjanmo replied:
퍼블리싱의 줄임말이군요 ㅎㅎ

jjanmo commented on hyunjin/client/public/readme/checked_item.png:
아래 적는 의견은 **진짜** 개인적인 의견이긴합니다.
저는 보통`README` 혹은 `마크다운 문서` 에서 넣는 이미지는 `screenshots` 이라는 폴더를 루트에 만들고(여기서는 이름명 폴더 바로 하위 겠죠) 거기에서 저장하는 편입니다. 

아마 현진님 의도는 클라이언트 관련 README이기때문에 클라이언트 폴더에서 README를 작성하신거 같고 그렇다면 같은 위계에서 
`screenshots` 라는 폴더가 있을거 같아요.
`public` 폴더는 코드에서 사용하는 정적인 리소스가 위치하는 공간이라고 생각하는게 일반적이긴해서, README의 screenshots 같은 경우는 따로 관리하는게 역할 분리의 측면에서 좋았던거 같습니다. 

별거 아니긴한데, 저도 예전에 많이 고민했던 부분이긴해서 혹시나 고민이 있으셨을까 해서 제가 경험한 의견 드려보아요 🙏🏻

guswls1846 replied:
> 아래 적는 의견은 **진짜** 개인적인 의견이긴합니다. 저는 보통`README` 혹은 `마크다운 문서` 에서 넣는 이미지는 `screenshots` 이라는 폴더를 루트에 만들고(여기서는 이름명 폴더 바로 하위 겠죠) 거기에서 저장하는 편입니다.

오! 맞아요 고민이 많았는데 좋은 방법인거 같아요 바로 적용해볼게요!

jjanmo commented on hyunjin/client/index.html:
```diff
@@ -5,9 +5,19 @@
-    <div id="app"></div>
-    <script type="module" src="/src/main.ts"></script>
+    <header class="header">
+      <a href="#" style="text-decoration: none"><h1 class="title">todos</h1></a>
```
스타일을 여기서만 inline으로 적어주신 이유가 있으실까요??

guswls1846 replied:
> 스타일을 여기서만 inline으로 적어주신 이유가 있으실까요??

 html이나 calss name은 예시 사이트를 가이드 삼고 있어서 개발자 도구의 elements에 있는 doc을 그대로 이용하고 있어요. 이유를 생각해보면 바닐라 스크립트를 사용하다보니 대부분의 dom 선택자를 class name 또는 id로 하기 때문에 굳이 접근할 필요가 없는 부분은 linline으로 처리하는거 같아요 어느정도 이 컨번션이 합당한거 같아 냅뒀습니다~

jjanmo replied:
아마도 작업하실 때 text-decorator 같은 경우는 보통 none으로 처리하여 a 태그에 있는 기본 밑줄을 없애는 경우가 대부분일거라는 가정을 하게 되는데요. 이런 경우 `global.css` 나 `reset.css` 에서 작업해주는게 좀 더 합리적일거라 생각해요. 
만약에 다른 a 태그가 생길 때마다 인라인으로 작성하는건 중복되는 코드가 발생하기도 할것 같구요,

개인적으로 바닐라자바스크립트로 작성하더라도 html, css, js 파일의 역할/기능을 잘 나누어서 사용하는 것이 나중을 위해서도 좋지않을까 하는 생각도 듭니다.

jjanmo commented on hyunjin/client/src/app/styles/index.css:
```diff
@@ -0,0 +1,43 @@
+:host {
+  display: block;
+  box-shadow: none !important;
+}
```
`:host`는 shadow dom를 선택하는 선택자라고 하네요.
혹시 shadow dom을 사용하실 예정이신가요??
(4장에서 Web Component API를 공부할 때 사용할거 같긴해요. 미리 사용해보시는 것도 👍🏻👍🏻)

guswls1846 replied:
shadow dom을 사용할거 같지만 해당 코드는  공용 styles을 담당하고 곳에 있을 필요는 없는거 같아요! 지우겠습니다

jjanmo commented on hyunjin/client/src/app/styles/index.css:
```diff
@@ -0,0 +1,43 @@
+  font-weight: 200;
+  text-align: center;
+  color: #b83f45;
+  -webkit-text-rendering: optimizeLegibility;
+  -moz-text-rendering: optimizeLegibility;
+  text-rendering: optimizeLegibility;
```
현진님 덕분에 해당 속성을 찾아봤습니다 ㅎㅎ

[참고]
- CSS의 text-rendering 속성 :  텍스트의 렌더링 방식을 제어하는데 사용
- `optimizeLegibility` : 글자 간격과 리가처(두 글자가 합쳐진 특수한 형태)를 사용하여 텍스트의 가독성을 향상시킴. 하지만 이 옵션은 렌더링 성능에 영향을 미칠 수 있으므로, 대량의 텍스트를 다루는 경우에는 신중하게 사용해야함

> 이라고 우리의 코파일럿이 알려주네요 ㅎㅎ

text-rendering이 CSS 표준에는 포함되어있지 않지만, [CanIuse](https://caniuse.com/?search=text-rendering)를 찾아보면 현재 대부분의 최신 브라우저에서 다 지원하고 있어서 -webkit 과 -moz를 붙이지 않아도 될 것 같습니다.

guswls1846 replied:
이부분은 제가 꼼꼼하지 못했네요.  다음부터 css 볼때도 내가 아는거와 모르는걸 확실하게 구분지으면서 작성 해봐야 되겠어요. 피드백 감사합니다!

jjanmo reviewed (approved):
고생하셨습니다. 👍🏻

저도 빨리 해야하는데 현진님 보고 화이팅해야겠습니다 ㅎㅎ

Commit:
move: readme 스크린샷 폴더 변경

Commit:
refactor: 미사용 중인 :host 선택자 제거

Commit:
refactor: text-rendering 접두어 제거

- text-rendering: optimizeLegibility를 사용하면 브라우저가 글자의 윤곽을 부드럽게 처리하고, 간격 조정을 통해 더욱 깔끔한 텍스트를 표시하게 됩니다

Commit:
feat: reset.css 파일 추가

Commit:
remove: 미사용 파일 제거

Commit:
rename: app/styles/index.css -> app/styles/global.css

- 해당 파일은 공통 css이므로 파일명을 global.css로 변경

Commit:
modify: reset.css button, input 추가

Commit:
feat: alias 설정

Commit:
refactor: footer 웹 컴포넌트로 변경

Commit:
refactor: header 웹 컴포넌트로 변경

Commit:
remove: .gitkeep파일 제거

Commit:
feat: shadow dom을 적용한 todo-input 컴포넌트 생성

Commit:
feat: todo-app 컴포넌트 생성

Commit:
feat: todo-list 퍼블리싱

Commit:
feat: todo-item 퍼블리싱

Commit:
refacor: todo-input features -> widgets로 이동

Commit:
refactor: 컴포넌트 이름 컨벤션 수정

- todoHeaderModel.TodoHeaderComponent() 이런 모양으로 호출이되서 class 이름을 단축했습니다.

Commit:
refactor: todo-header, todo-footer를 todo-app 내부로 이동

Commit:
feat: todo-bottombar 퍼블리싱

Commit:
modify: todo-topbar에서 체크박스 보이게 수정

Commit:
Merge branch 'publishing'
